restord/ImageSessionServer.m

```diff
@@ -399,4 +426,57 @@ - (void)blessMountURL:(NSURL *)mountURL {
   if (bless.terminationStatus != 0) NSLog(@"%@ Failed to bless: %@", self, mountURL.path);
 }
 
+- (int)runPostScriptWithMountURL:(NSURL *)mountURL error:(NSError **)error {
+  if (!self.image.postScript) return 0;
+
+  NSString *uuid = [[NSProcessInfo processInfo] globallyUniqueString];
+  NSString *path = [NSTemporaryDirectory() stringByAppendingPathComponent:uuid];
+  NSURL *directoryURL = [NSURL fileURLWithPath:path isDirectory:YES];
+
+  NSError *err;
+  NSFileManager *fm = [NSFileManager defaultManager];
+  if (![fm createDirectoryAtURL:directoryURL
+    withIntermediateDirectories:YES
+                     attributes:nil
+                          error:&err]) {
+    if (error) *error = err;
+    return 1;
+  }
+
+  NSString *postscript = [directoryURL.path stringByAppendingPathComponent:@"postscript"];
+  if (![self.image.postScript writeToFile:postscript
+                               atomically:YES
+                                 encoding:NSUTF8StringEncoding
+                                    error:&err]) {
+    if (error) *error = err;
+    return 1;
+  }
+
+  if (![fm setAttributes:@{ NSFilePosixPermissions : @0755 } ofItemAtPath:postscript error:&err]) {
+    if (error) *error = err;
+    return 1;
+  }
+
+  NSTask *script = [[NSTask alloc] init];
+  script.launchPath = postscript;
+  script.arguments = @[ mountURL.path ];
+  script.standardError = [NSPipe pipe];
+
+  NSLog(@"%@ Launching post script: %@", self, postscript);
+
+  [script launch];
+  [script waitUntilExit];
+
+  NSData *data = [script.standardError fileHandleForReading].availableData;
+  NSString *standardError = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+  if (standardError.length && error) {
+    NSDictionary *info = @{ NSLocalizedDescriptionKey: standardError };
+    *error = [NSError errorWithDomain:@"com.google.corp.restord" code:666 userInfo:info];
+  }
+
+  [fm removeItemAtPath:postscript error:NULL];
```

Comment:
Remove the parent directory too (or instead of)?

Reply:
Done.